File: src/main/java/io/github/starwishsama/namelessbot/commands/InfoCommand.java
package io.github.starwishsama.namelessbot.commands;

import cc.moecraft.icq.command.CommandProperties;
import cc.moecraft.icq.command.interfaces.EverywhereCommand;
import cc.moecraft.icq.event.events.message.EventMessage;
import cc.moecraft.icq.user.User;
import io.github.starwishsama.namelessbot.BotConstants;
import io.github.starwishsama.namelessbot.objects.user.BotUser;
import io.github.starwishsama.namelessbot.utils.BotUtils;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public class InfoCommand implements EverywhereCommand {
    @Override
    public CommandProperties properties(){
        return new CommandProperties("info", "查询", "查", "cx");
    }

    @Override
    public String run(EventMessage event, User sender, String command, ArrayList<String> args) {
        try {
            if (args.size() == 0) {
                if (BotUser.isUserExist(sender.getId())) {
                    BotUser user = BotUser.getUser(sender.getId());
                    String reply = "[CQ:at,qq=" + sender.getId() + "]\n积分: " + String.format("%.1f", Objects.requireNonNull(user).getCheckInPoint())
                            + "\n累计连续签到了 " + user.getCheckInTime() + " 天"
                            + "\n上次签到于: " + new SimpleDateFormat("yyyy-MM-dd").format(user.getLastCheckInTime().getTime())
                            + "\n你的权限组: " + user.getLevel()
                            + "\n占卜次数: " + user.getRandomTime();
                    if (user.getBindServerAccount() != null) {
                        reply = reply + "绑定的游戏账号是: " + user.getBindServerAccount();
                    }
                    return reply;
                } else {
                    return BotUtils.getLocalMessage("msg.bot-prefix") + "你还没有签到过哦, 使用 /qd 签到一下吧~";
                }
            } else if (args.size() == 1 && args.get(0).equalsIgnoreCase("排行") || args.get(0).equalsIgnoreCase("ph")) {
                List<BotUser> users = (List<BotUser>) BotConstants.users;
                users.sort((u1, u2) -> (int) (u2.getCheckInPoint() - u1.getCheckInPoint()));
                StringBuilder sb = new StringBuilder();
                sb.append("积分排行榜").append("\n");

                for (int i = 0; i < 10; i++){
                    sb.append(i + 1).append(" ")
                            .append(event.getHttpApi().getStrangerInfo(users.get(i).getUserQQ()).getData().getNickname())
                            .append(" ").append(String.format("%.1f", users.get(i).getCheckInPoint())).append("\n");
                }

                return sb.toString().trim();
            }
        } catch (Exception e){
            e.printStackTrace();
        }
        return null;
    }
}

File: src/main/kotlin/io/github/starwishsama/nbot/commands/subcommands/InfoCommand.kt
package io.github.starwishsama.nbot.commands.subcommands

import io.github.starwishsama.nbot.BotConstants
import io.github.starwishsama.nbot.BotInstance
import io.github.starwishsama.nbot.commands.CommandProps
import io.github.starwishsama.nbot.commands.interfaces.UniversalCommand
import io.github.starwishsama.nbot.enums.UserLevel
import io.github.starwishsama.nbot.objects.BotUser
import io.github.starwishsama.nbot.util.BotUtils.getLocalMessage
import net.mamoe.mirai.message.ContactMessage
import net.mamoe.mirai.message.GroupMessage
import net.mamoe.mirai.message.data.*
import java.time.format.DateTimeFormatter

class InfoCommand : UniversalCommand {
    override suspend fun execute(message: ContactMessage, args: List<String>, user: BotUser): MessageChain {
        try {
            if (args.size == 1) {
                return run {
                    var reply =
                            "\n积分: "+ String.format("%.1f", user.checkInPoint) +
                                    "\n累计连续签到了 " + user.checkInTime.toString() + " 天" + "\n上次签到于: " +
                                    user.lastCheckInTime.format(DateTimeFormatter.ofPattern("yyyy-MM-dd")).toString() +
                                    "\n权限组: " + user.level.toString() +
                                    "\n命令条数: " + user.randomTime
                    if (user.bindServerAccount != null) {
                        reply = reply + "绑定的游戏账号是: " + user.bindServerAccount
                    }

                    if (message is GroupMessage){
                        message.sender.at() + reply.toMessage().asMessageChain()
                    } else {
                        reply.toMessage().asMessageChain()
                    }
                }
            } else if (args.size == 2 && args[1].contentEquals("排行") || args[1].contentEquals("ph")) {
                val users = BotConstants.users
                users.sortedByDescending { it.checkInPoint }
                val sb = StringBuilder()
                sb.append("积分排行榜").append("\n")
                return if (users.size > 9) {
                    for (i in 0..9) {
                        sb.append(i + 1).append(" ")
                            .append(message.sender.nick)
                            .append(" ").append(String.format("%.1f", users[i].checkInPoint)).append("\n")
                    }
                    (sb.toString().trim { it <= ' ' }).toMessage().asMessageChain()
                } else {
                    "数据不足".toMessage().asMessageChain()
                }
            } else {
                return getHelp().toMessage().asMessageChain()
            }
        } catch (e: Exception) {
            BotInstance.logger.error(e)
        }
        return EmptyMessageChain
    }

    override fun getProps(): CommandProps = CommandProps("info", arrayListOf("cx", "查询"), "nbot.commands.info", UserLevel.USER)
    override fun getHelp(): String = """
        ======= 命令帮助 =======
        /cx 查询自己的积分信息
        /cx ph 查询积分排行榜
    """.trimIndent()

}
